src/services/getEditsForFileRename.ts

```diff
@@ -47,23 +41,61 @@ namespace ts {
                 const resolved = host.resolveModuleNames
                     ? host.getResolvedModuleWithFailedLookupLocationsFromCache && host.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName)
                     : program.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName);
-                if (resolved && contains(resolved.failedLookupLocations, oldFilePath)) {
-                    result.push({ sourceFile, toUpdate: importStringLiteral });
+                const updated = resolved && firstDefined(resolved.failedLookupLocations, path => pathUpdater(path, importStringLiteral.text, /*isImport*/ true));
+                if (updated !== undefined) {
+                    changeTracker.replaceRangeWithText(sourceFile, createStringRange(importStringLiteral, sourceFile), updated);
                 }
             }
         }
-        return result;
     }
 
-    function getPathUpdater(oldFilePath: string, newFilePath: string, host: LanguageServiceHost): (oldPath: string) => string | undefined {
+    /**
+     * @param oldFullPath Absolute path to a failed lookup location
+     * @param oldRelPath Actual import text
+     */
+    type PathUpdater = (oldFullPath: string, oldRelPath: string, isImport: boolean) => string | undefined;
+    function getPathUpdater(oldFileOrDirPath: string, newFileOrDirPath: string, host: LanguageServiceHost): PathUpdater {
         // Get the relative path from old to new location, and append it on to the end of imports and normalize.
-        const rel = getRelativePathFromFile(oldFilePath, newFilePath, createGetCanonicalFileName(hostUsesCaseSensitiveFileNames(host)));
-        return oldPath => {
-            if (!pathIsRelative(oldPath)) return;
-            return ensurePathIsNonModuleName(normalizePath(combinePaths(getDirectoryPath(oldPath), rel)));
+        const rel = getRelativePathFromFile(oldFileOrDirPath, newFileOrDirPath, createGetCanonicalFileName(hostUsesCaseSensitiveFileNames(host)));
+        return (fullPath, relPath, isImport) => {
+            if (fullPath === oldFileOrDirPath) {
+                const newRelPath = pathIsRelative(relPath)

```

Comment:
Nit: Pls rename relPath to say textInFile or something like that since it could be non relatvie?